tests/NCronJob.Tests/TestHelper.cs

```diff
@@ -140,10 +140,9 @@ protected async Task WaitUntilConditionIsMet(
                 return;
             }
 
-            var delay = TimeSpan.FromMicroseconds(10);
-            FakeTimer.Advance(delay);
+            FakeTimer.Advance(TimeSpan.FromSeconds(1));
 
-            await Task.Delay(delay, CancellationToken);
+            await Task.Delay(TimeSpan.FromMilliseconds(20));
```

Comment:
@linkdotnet Since your approval, I shoved this additional change in. 

Indeed, #199 put under the light some time related issues (that may find their root cause in the TimeProvider changes from #178)

Reply:
Interesting - I kind of understand the increase in time for the delay - but wouldn't it still make sense to pass in the `CancellationToken`.

I will let the CI run a few times to see if those tests are affected by this change are more stable.

Reply:
Running 6x without issue so far